Kernel/Syscalls/socket.cpp

```diff
@@ -199,32 +198,47 @@ ErrorOr<FlatPtr> Process::sys$sendmsg(int sockfd, Userspace<const struct msghdr*
             Thread::current()->send_signal(SIGPIPE, &Process::current());
         return EPIPE;
     }
-    auto data_buffer = TRY(UserOrKernelBuffer::for_user_buffer((u8*)iovs[0].iov_base, iovs[0].iov_len));
 
-    while (true) {
-        while (!description->can_write()) {
-            if (!description->is_blocking()) {
-                return EAGAIN;
+    Checked<ssize_t> total_length = 0;
+    for (auto const& iov : iovs)
+        total_length += iov.iov_len;
+
+    if (total_length.has_overflow())
+        return EINVAL;
+
+    size_t nwrite = 0;
+    for (auto const& iov : iovs) {
+        auto data_buffer = TRY(UserOrKernelBuffer::for_user_buffer((u8*)iov.iov_base, iov.iov_len));
+
+        while (true) {
+            while (!description->can_write()) {
+                if (!description->is_blocking()) {
+                    return EAGAIN;
+                }
+
+                auto unblock_flags = Thread::FileBlocker::BlockFlags::None;
+                if (Thread::current()->block<Thread::WriteBlocker>({}, *description, unblock_flags).was_interrupted()) {
+                    return EINTR;
+                }
+                // TODO: handle exceptions in unblock_flags
             }
 
-            auto unblock_flags = Thread::FileBlocker::BlockFlags::None;
-            if (Thread::current()->block<Thread::WriteBlocker>({}, *description, unblock_flags).was_interrupted()) {
-                return EINTR;
+            auto bytes_sent_or_error = socket.sendto(*description, data_buffer, iov.iov_len, flags, user_addr, addr_length);
+            if (bytes_sent_or_error.is_error()) {
+                if ((flags & MSG_NOSIGNAL) == 0 && bytes_sent_or_error.error().code() == EPIPE)
+                    Thread::current()->send_signal(SIGPIPE, &Process::current());
+                return bytes_sent_or_error.release_error();
             }
-            // TODO: handle exceptions in unblock_flags
-        }
 
-        auto bytes_sent_or_error = socket.sendto(*description, data_buffer, iovs[0].iov_len, flags, user_addr, addr_length);
-        if (bytes_sent_or_error.is_error()) {
-            if ((flags & MSG_NOSIGNAL) == 0 && bytes_sent_or_error.error().code() == EPIPE)
-                Thread::current()->send_signal(SIGPIPE, &Process::current());
-            return bytes_sent_or_error.release_error();
+            auto bytes_sent = bytes_sent_or_error.release_value();
+            if (bytes_sent) {
```

Comment:
```suggestion
            if (bytes_sent > 0) {
```

Reply:
I've addressed these suggestions in my branch and force pushed the changes. Thank you.